### ISSUE TITLE
[Feature requests] Make Matomo users level above Matomo account levels so Matomo users can access multiple accounts without creating one user per Matomo account

### ISSUE THREAD

**Issue by SW-Vincent** [piwik/piwik#21988]
I work in an agency with about 20 Matomo accounts, so it'd be really nice if i could use as one user create / access multiple Matomo domains.

## Summary

I think i am not the only one with this need, as it is pretty common for businesses to ask an agency to handle Web Analytics.
Being able to have some user account (would it be shared accounts like GA or manager accounts like Google Ads) would have several benefits : 
 - for agencies : a more simple way to handle multiple accounts, better view over facturation state, etc. ;
 - for Matomo / agencies : can link in-app contact forms to a number of account / cumulative generated revenue if that is a criterion for prioritizing answers ;
 - for Matomo : better view of customer structures (segmentations between cutomers using Matomo on their own vs customer using Matomo through agencies), that would also help if Matomo ever intents to leverage its services through agencies (or by any means push some notifications to some specific users only, "Join the Matomo Feedback Panel" may be one) ;

I do not know how it would apply on wordpress / on-premise versions.

## Your Environment

* Matomo Version: Mostly cloud accounts
* PHP Version:
* Server Operating System:
* Additionally installed plugins:

**Comment by michalkleiner** [comment]
Hi @SW-Vincent and thanks for the suggestion! I'm not 100% sure I understand the structure of the accounts and which type could access which instance/which measurables, could you maybe do a little diagram or try to explain it in different terms? Since you mention not being sure about on-premise, would this be for Matomo cloud? And wanting a single account to access multiple Matomo instances? They are still technically separate instances, so it might be possible using some login federation service/LDAP and appropriate roles. But happy for you to clarify if I'm getting it right. Thank you!

**Comment by SW-Vincent** [comment]
Indeed, the whole point -in my opinion- is either : 
 - wanting a single account to access several Matomo instances 
 OR
 - having a new level of instances : Manager instances that would have access to several instances

**Comment by michalkleiner** [comment]
Right, thanks for the clarification. I'll assign the enhancement suggestion to the product team for consideration. Thank you for the suggestion!

**Comment by sgiehl** [comment]
This is actually something we cannot implement in core or for Wordpress. Each Matomo instance uses it's own database (even on cloud). And an instance can't access the database of another instance. So to implement something like this, it requires a global database, every instance additionally has access to. 
It could be something we can implement on cloud. For on premise/Wordpress I guess a separate plugin would be needed to handle that.
